src/fixit/config.py

```diff
@@ -156,26 +224,82 @@ def merge_configs(
 
     Assumes raw_configs are given in order from highest to lowest priority.
     """
-    kwargs: Dict[str, Any] = {
-        "root": None,
-    }
+
+    enable_rules: Set[str] = set()
+    disable_rules: Set[str] = set()
+    local_paths: List[Path] = []
+    rule_options: RuleConfigs = {}
+
+    def process_subpath(
+        subpath: Path,
+        *,
+        enable: Sequence[str] = (),
+        disable: Sequence[str] = (),
+        options: RuleConfigs = frozendict(),
+    ):
+        subpath = subpath.resolve()
+        if not path.is_relative_to(subpath):
+            print(f"ignoring {subpath}")
+            return
+
+        for rule in enable:
+            if rule.startswith("."):
+                if not local_paths or local_paths[0] != subpath:
+                    local_paths.insert(0, subpath)
+
+            enable_rules.add(rule)
+            disable_rules.discard(rule)
+
+        for rule in disable:
+            enable_rules.discard(rule)
+            disable_rules.add(rule)
+
+        if options:
+            rule_options.update(options)
 
     for config in reversed(raw_configs):
-        if kwargs["root"] is None:
-            kwargs["root"] = config.path.parent
+        if root is None:
+            root = config.path.parent
+
+        data = config.data
+        if data.pop("root", False):
+            root = config.path.parent
+
+        process_subpath(
+            config.path.parent,
+            enable=get_sequence(config, "enable"),
+            disable=get_sequence(config, "disable"),
+            options=get_options(config, "options"),
+        )
+
+        for override in get_sequence(config, "overrides"):
+            if not isinstance(override, dict):
+                raise ConfigError("'overrides' requires array of tables", config=config)
```

Comment:
Hmm, maybe better to do validation in some other place? Like when we're loading the raw configs. It'd be also good to validate that `disable_rules` can't contain local rules (since they aren't supported by the code here)

Reply:
Local rules in general aren't yet supported, but as a user, I would be disappointed if there weren't a way to disable local rules once they work.